### PR TITLE
Add `doc_type` field in `record` model

### PR DESCRIPTION
**Pull request by moesoha**
Related to `feature/homework` (#460)

https://github.com/vijos/vj4/pull/460#discussion_r259571773

### CODE REVIEW

File: vj4/handler/judge.py [hunk 1]
```diff
@@ -47,7 +48,7 @@
     if accept:
       post_coros.append(_send_ac_mail(handler, rdoc))
     if rdoc['tid']:
-      post_coros.append(contest.update_status(rdoc['domain_id'], rdoc['tid'], rdoc['uid'],
+      post_coros.append(contest.update_status(rdoc['domain_id'], rdoc['doc_type'] if 'doc_type' in rdoc else document.TYPE_CONTEST, rdoc['tid'], rdoc['uid'],
```
Comment (twd2) [inline vj4/handler/judge.py]: `rdoc.get('doc_type', document.TYPE_CONTEST)`

File: vj4/handler/record.py [hunk 2]
```diff
@@ -23,8 +23,7 @@
 
 class RecordVisibilityMixin(contest.ContestVisibilityMixin):
   async def rdoc_contest_visible(self, rdoc):
-    # TODO(soha): use contest.get
-    tdoc = await document.get(rdoc['domain_id'], {'$in': [document.TYPE_CONTEST, document.TYPE_HOMEWORK]}, rdoc['tid'])
+    tdoc = await contest.get(rdoc['domain_id'], rdoc['doc_type'] if 'doc_type' in rdoc else document.TYPE_CONTEST, rdoc['tid'])
```
Comment (twd2) [inline vj4/handler/record.py]: ditto

File: vj4/model/record.py [hunk 3]
```diff
@@ -34,6 +34,7 @@
          'lang': lang,
          'code': code,
          'tid': tid,
+         'doc_type': doc_type,
```
Comment (twd2) [inline vj4/model/record.py]: `ttype`?

**Comment by breeswish** [comment]
Should we perform database updates for existing data?

**Comment by twd2** [comment]
`ttype` is optional.

**Pull request merged**